src/main/java/duke/data/TaskList.java

```diff
@@ -0,0 +1,83 @@
+package duke.data;
+
+import java.util.ArrayList;
+
+import duke.task.Task;
+
+/**
+ * Represents the list of tasks that the user writes to.
+ */
+public class TaskList {
+
+    protected ArrayList<Task> taskList;
+
+    /**
+     * Constructor for the TaskList object.
+     *
+     * @param taskList an ArrayList of tasks representing the saved task from a previous program execution
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Default constructor for the TaskList.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Returns the number of tasks in the taskList.
+     *
+     * @return the number of tasks in the taskList
+     */
+    public int getLength() {
```

Comment:
getAmountOfTasks() or getSize() maybe more meaningful. 
Overall you did a really good job. Your code is very neat and elegant with very few violations of naming rules.